feat(metadata-linter): impl new linter (#5610)
canonical/snapcraft: docs/how-to/debugging/disable-a-linter.rst

      ignore:
        - classic
        - library
        - metadata

> .. _how-to-disable-a-linter-ignore-specific-files-and-field:

Reviewer: ```suggestion
.. _how-to-disable-a-linter-ignore-specific-files:
```

Instead of altering the reference name, let's add a new one after it, and name it after the modified heading. Both refs will then work.

Ignore specific files

> ---------------------

Reviewer: This should become "Ignore files and keys".

``lint.ignore`` key. The path is relative to the snap directory tree, and supports

> wildcard characters (*).

Reviewer: In this paragraph, please describe how for one of the linters, keys can also be ignored.

Author: Sorry, didn't understand this. Can you kindly explain this about which linter you're asking this change for? The metadata-linter doesn't have any {key: value} pair, it's just keys, and I think I have added one in the doc.

Reviewer: I was thinking of something like this:

```rst
If you need to ignore specific elements in your project, add them to the ``lint.ignore``
key. Each entry of the key represents a built-in linter, and lists the elements you want
that linter to ignore. For file paths, make them relative to the snap directory tree,
and use wildcards (*) to match multiple files and directories. For project file keys,
match the name of the key.
```

In the following example, the ``classic`` linter is disabled entirely, the
``library`` linter won't run for the files in ``usr/lib`` that match the specified

> pattern and the ``title`` field will be ignored by the ``metadata`` linter:

Reviewer: ```suggestion
pattern, and the ``title`` field will be ignored by the ``metadata`` linter:
```

        - classic
        - library:
          - usr/lib/**/libfoo.so*

.. _how-to-disable-a-linter-ignore-specific-keys-of-metadata-linter:

Ignore specific keys of the metadata linter
-------------------------------------------

You can disable the metadata linter for specific keys by listing them in the

> ``lint.ignore`` key. The keys are from the top level keys of a snpacraft.yaml file.

Reviewer 2: I'm glad it caught this, will you change it to this? This is the preferred terminology in the docs:

```suggestion
``lint.ignore`` key. The keys are from the top level keys of a project file.
```

In the following example, the ``title`` and ``issues`` keys are ignored by the
metadata linter.

.. code-block:: yaml
    :caption: snapcraft.yaml

    lint:
      ignore:
        - title

>         - issues

Author: Yep, it was right
